src/pool/validate.rs

```diff
@@ -279,7 +279,14 @@ where
             return TransactionValidationOutcome::Invalid(transaction, err);
         }
 
-        let account = match self.client.basic_ref(transaction.sender()) {
+        let handle = tokio::runtime::Handle::current();
+        let block = match tokio::task::block_in_place(|| handle.block_on(self.provider.block_number())) {
+            Ok(b) => b,
+            Err(err) => return TransactionValidationOutcome::Error(*transaction.hash(), Box::new(err)),
+        };
+        let db = EthDatabase::new(Arc::new(&self.provider), BlockId::Number(BlockNumberOrTag::Number(block.to())));
```

Comment:
Here I think that you can directly do:

```rust
let db = EthDatabase::new(Arc::new(&self.provider), BlockId::from(block.to()));
```

no?

Reply:
ah you might be right, let me change, need to fix the tests anyway